netty-utils/src/test/java/org/asynchttpclient/netty/util/ByteBufUtilsTests.java

```diff
@@ -0,0 +1,71 @@
+package org.asynchttpclient.netty.util;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.nio.charset.Charset;
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.testng.internal.junit.ArrayAsserts;
+
+public class ByteBufUtilsTests {
+
+    @Test
+    public void testByteBuf2Bytes() {
+        ByteBuf byteBuf = Unpooled.copiedBuffer(new byte[]{'f', 'o', 'o'});
+        ArrayAsserts.assertArrayEquals(new byte[]{'f', 'o', 'o'},
+                ByteBufUtils.byteBuf2Bytes(byteBuf));
+
+        ByteBuf buf = Unpooled.buffer();
+        ArrayAsserts.assertArrayEquals(new byte[]{},
+                ByteBufUtils.byteBuf2Bytes(buf));
+    }
+
+    @Test
+    public void testByteBuf2String1() {
+        ByteBuf byteBuf = Unpooled.copiedBuffer(new byte[]{'f', 'o', 'o'});
+        Charset charset = Charset.forName("US-ASCII");
+
+        Assert.assertEquals(
+                ByteBufUtils.byteBuf2String(charset, byteBuf), "foo");
+    }
+
+    @Test
+    public void testByteBuf2String2() {
+        ByteBuf byteBuf1 = Unpooled.copiedBuffer(new byte[]{'f'});
+        ByteBuf byteBuf2 = Unpooled.copiedBuffer(new byte[]{'o', 'o'});
+
+        Assert.assertEquals(ByteBufUtils.byteBuf2String(
+                Charset.forName("ISO-8859-1"), byteBuf1, byteBuf2), "foo");
+    }
+
+    @Test
+    public void testByteBuf2Chars1() {
+        ByteBuf byteBuf1 = Unpooled.copiedBuffer(new byte[]{});
+        ByteBuf byteBuf2 = Unpooled.copiedBuffer(new byte[]{'o'});
+
+        ArrayAsserts.assertArrayEquals(new char[]{}, ByteBufUtils
+                .byteBuf2Chars(Charset.forName("ISO-8859-1"), byteBuf1));
```

Comment:
Why use ISO-8859-1 and not US-ASCII?

Reply:
As I was trying to cover the branch `decodeChars(buf, charset)` in the ` byteBuf2Chars(Charset charset, ByteBuf buf)` method.